lib/iris/tests/test_grib_save.py

```diff
@@ -40,84 +40,89 @@
 
 @tests.skip_data
 @tests.skip_grib
-class TestLoadSave(tests.IrisTest):
-    # load and save grib
+class TestLoadSave(tests.TestGribMessage):
 
     def setUp(self):
-        iris.fileformats.grib.hindcast_workaround = True
-        self.problem_gribapi_ver = self._problem_gribapi_ver()
-
-    def tearDown(self):
-        iris.fileformats.grib.hindcast_workaround = False
-
-    def _problem_gribapi_ver(self):
-        # The difference we're checking for here arrived in v1.12.0
-        try:
-            # gribapi v1.9.16 has no __version__ attribute.
-            gribapi_ver = gribapi.__version__
-        except AttributeError:
-            gribapi_ver = gribapi.grib_get_api_version()
-        return StrictVersion(gribapi_ver) < StrictVersion('1.12.0')
-
-    def save_and_compare(self, source_grib, reference_text):
-        """Load and save grib data, generate diffs, compare with expected diffs."""
-
-        # load and save from Iris
-        cubes = iris.load(source_grib)
-
-        saved_grib = iris.util.create_temp_filename(suffix='.grib2')
-        iris.save(cubes, saved_grib)
-
-        # missing reference? (the expected diffs between source_grib and saved_grib)
-        if not os.path.exists(reference_text):
-            warnings.warn("Creating grib compare reference %s" % reference_text)
-            os.system("grib_compare %s %s > %s" % (source_grib, saved_grib, reference_text))
-
-        # generate and compare diffs
-        compare_text = iris.util.create_temp_filename(suffix='.grib_compare.txt')
-        os.system("grib_compare %s %s > %s" % (source_grib, saved_grib, compare_text))
-        self.assertTextFile(compare_text, reference_text, "grib_compare output")
-
-        os.remove(saved_grib)
-        os.remove(compare_text)
+        self.skip_keys = []
+        if gribapi.__version__ < '1.13':
+            self.skip_keys = ['g2grid', 'latitudeOfLastGridPointInDegrees',
+                              'iDirectionIncrementInDegrees',
+                              'longitudeOfLastGridPointInDegrees',
+                              'latLonValues', 'distinctLatitudes',
+                              'distinctLongitudes', 'lengthOfHeaders',
+                              'values']
 
     def test_latlon_forecast_plev(self):
         source_grib = tests.get_data_path(("GRIB", "uk_t", "uk_t.grib2"))
-        if self.problem_gribapi_ver:
-            result_file = "latlon_forecast_plev.grib_compare.pre1-12-0.txt"
-        else:
-            result_file = "latlon_forecast_plev.grib_compare.post1-12-0.txt"
-        reference_text = tests.get_result_path(("grib_save", result_file))
-        self.save_and_compare(source_grib, reference_text)
+        cubes = iris.load(source_grib)
+        with self.temp_filename(suffix='.grib2') as temp_file_path:
+            iris.save(cubes, temp_file_path)
+            expect_diffs = {2: ('exists', None),
+                            'totalLength': (4837, 4832),
```

Comment:
We are more able to manage them in this way.  We can record known diffs and, if required, explicitly skip keys if there are grib api problems

the key difference is that we are looking on an element basis, rather than relying on the terminal output of grib_compare, which is aimed at humans, not machines